iree/compiler/Dialect/Stream/Transforms/test/propagate_subviews.mlir

```diff
@@ -0,0 +1,148 @@
+// RUN: iree-opt -split-input-file -iree-stream-propagate-subviews %s | IreeFileCheck %s
+
+// Tests that resource global loads also load all the subview params.
+//
+// This rotates subviews through stores and into loads.
+
+//      CHECK: util.global private mutable @constantGlobal : !stream.resource<constant>
+// CHECK-NEXT: util.global private mutable @constantGlobal__storage_size : index
+// CHECK-NEXT: util.global private mutable @constantGlobal__offset : index
+// CHECK-NEXT: util.global private mutable @constantGlobal__length : index
+util.global private mutable @constantGlobal : !stream.resource<constant>
+
+// CHECK-LABEL: @globalLoad
+func @globalLoad() {
+  // CHECK-NEXT: %[[RESOURCE:.+]] = util.global.load @constantGlobal : !stream.resource<constant>
+  // CHECK-NEXT: %[[STORAGE_SIZE:.+]] = util.global.load @constantGlobal__storage_size : index
+  // CHECK-NEXT: %[[OFFSET:.+]] = util.global.load @constantGlobal__offset : index
+  // CHECK-NEXT: %[[LENGTH:.+]] = util.global.load @constantGlobal__length : index
+  // CHECK: %[[SUBVIEW:.+]] = stream.resource.subview %[[RESOURCE]][%[[OFFSET]]] : !stream.resource<constant>{%[[STORAGE_SIZE]]} -> !stream.resource<constant>{%[[LENGTH]]}
+  %0 = util.global.load @constantGlobal : !stream.resource<constant>
+  // CHECK-NEXT: util.do_not_optimize(%[[SUBVIEW]])
+  util.do_not_optimize(%0) : !stream.resource<constant>
+  return
+}
+
+// -----
+
+// Tests that resource global stores consume their incoming subviews.
+//
+// This rotates subviews through stores and into loads.
```

Comment:
I don't see any "subview" ops in the original or checked IR for this test case. Are they produced then removed? If so, maybe add a `CHECK-NOT`?

Reply:
ah, they aren't actually needed - this pass is just expanding things unconditionally and isn't reliant on subview ops being in the IR - this way if we add subview ops during this pass or later on they are handled generically and the pass is safe to run any number of times

Reply:
(you can think of this like if you had a pass that widened all function arguments in C from whatever they were to int64_t - you'd insert the casts to/from int64_t unconditionally and then allow canonicalization to say "wait, this is already int64_t" and such)